features/smoke_tests/handled.feature

```diff
@@ -0,0 +1,164 @@
+Feature: Handled smoke tests
+
+Scenario: Notify caught Java exception with default configuration
+    When I run "HandledJavaSmokeScenario"
+    And I wait to receive a request
+    Then the request is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier
+
+    # Exception details
+    And the payload field "events" is an array with 1 elements
+    And the exception "errorClass" equals "java.lang.RuntimeException"
+    And the exception "message" equals "HandledJavaSmokeScenario"
+    And the exception "type" equals "android"
+    And the payload field "events.0.exceptions.0.stacktrace" is a non-empty array
```

Comment:
It'd be nice to have some more validation of the stacktrace for both JVM/native reports, as both are serialized quite differently. Stacktraces tend to be quite brittle and change when scenario code alters, so it would be fine for now just to check that fields aren't null..